### PR TITLE
Move globalSettings to projectSettings (fix #468)

### LINKED ISSUE #468: publishSigned no longer works
```
sbt diffJVM/publishSigned
... 
[error] (diffJVM/*:publishSigned) java.io.IOException: PUT operation to URL https://oss.sonatype.org/content/repositories/snapshots/ch/epfl/scala/scalafix-diff_2.12/0.5.6/scalafix-diff_2.12-0.5.6-sources.jar failed with status code 400: Bad Request
```

Can you reproduce @MasseGuillaume ? Any chance this is related to the build refactoring?

### PR DESCRIPTION
**Pull request by MasseGuillaume**
for some reason sbt is not picking up `globalSettings`

### CODE REVIEW

File: project/ScalafixBuild.scala [hunk 1]
```diff
@@ -247,15 +238,22 @@ object ScalafixBuild extends AutoPlugin with GhpagesKeys {
...
-  )
-
-  private val PreviousScalaVersion = Map(
-    "2.11.12" -> "2.11.11",
-    "2.12.4" -> "2.12.3"
-  )
-
-  override def projectSettings: Seq[Def.Setting[_]] = List(
+    ),
+    publishTo := {
```
Comment (olafurpg) [inline project/ScalafixBuild.scala]: What happens if you move these to `buildSettings`?

Comment (MasseGuillaume) [reply project/ScalafixBuild.scala]: What is the difference?

Comment (olafurpg) [reply project/ScalafixBuild.scala]: reload performance, `projectSettings` are evaluated for every project while `buildSettings` are only evaluated once for the build. scalafix has 64 projects, and adding 2.13/native support would increase that number even more.

Comment (MasseGuillaume) [reply project/ScalafixBuild.scala]: Done.

File: project/ScalafixBuild.scala [hunk 2]
```diff
@@ -160,62 +160,12 @@ object ScalafixBuild extends AutoPlugin with GhpagesKeys {
   lazy val adhocRepoCredentials = sys.props("scalafix.repository.credentials")
   lazy val isCustomRepository = adhocRepoUri != null && adhocRepoCredentials != null
 
-  override def globalSettings: Seq[Def.Setting[_]] = List(
```
Comment (olafurpg) [inline project/ScalafixBuild.scala]: Are all of these changes necessary to fix publishing? I thought it was only publishTo that was not picked up. I don't have a strong opinion, this change is fine, just trying to learn

Comment (MasseGuillaume) [reply project/ScalafixBuild.scala]: yeah publishTo, scmInfo all of this has an impact on publishSigned.

**Pull request merged**